packages/mco/utils/managed-cluster-view.ts

```diff
@@ -0,0 +1,144 @@
+import { NotFoundError } from '@odf/shared/utils/error/http-error';
+import {
+  getAPIVersionForModel,
+  k8sCreate,
+  k8sDelete,
+  k8sGet,
+  K8sResourceCommon,
+} from '@openshift-console/dynamic-plugin-sdk';
+import { TFunction } from 'react-i18next';
+import { v4 as uuidv4 } from 'uuid';
+import { ACMManagedClusterViewModel } from '../models';
+import { ACMManagedClusterViewKind } from '../types';
+
+export const deleteManagedClusterView = (
+  viewName: string,
+  clusterName: string
+) =>
+  k8sDelete({
+    model: ACMManagedClusterViewModel,
+    resource: { metadata: { name: viewName, namespace: clusterName } },
+  }).catch((err) => {
+    if (err instanceof NotFoundError && err.message !== 'NotFound') throw err;
+  });
+
+export const getManagedClusterView = (
+  viewName: string,
+  clusterName: string
+): Promise<ACMManagedClusterViewKind> =>
+  k8sGet({
+    model: ACMManagedClusterViewModel,
+    name: viewName,
+    ns: clusterName,
+  });
+
+export const pollManagedClusterView = (
+  viewName: string,
+  clusterName: string,
+  t: TFunction
+): Promise<PollManagedClusterViewResult> => {
+  let retries = process.env.NODE_ENV === 'development' ? 0 : 20;
```

Comment:
@GowthamShanmugam this won't work at runtime (and in my `Object Browser on consumer clusters` PR that I'll raise ASAP `process.env` references in our browser code will be deleted @bipuladh @SanjalKatiyar ). For a quick workaround use this instead (like in `noobaa-context.tsx`):
```suggestion
  let retries = window.location.hostname.includes('localhost') ? 0 : 20;
```

Reply:
btw do we really need this condition ?? what's wrong with `retries = 20` in dev mode ??